pkg/netpol/eval/internal/k8s/adminnetpol.go

```diff
@@ -66,13 +230,50 @@ func doesPodsFieldMatchPeer(pods *apisv1a.NamespacedPod, peer Peer) (bool, error
 	return nsSelector.Matches(labels.Set(peer.GetPeerNamespace().Labels)) && podSelector.Matches(labels.Set(peer.GetPeerPod().Labels)), nil
 }
 
+// doesNetworksFieldMatchPeer checks if the given peer matches the networks field.
+// returns true if the peer's IPBlock is contained within any of the cidr-s in networks field
+func doesNetworksFieldMatchPeer(networks []apisv1a.CIDR, peer Peer) (bool, error) {
+	if peer.GetPeerIPBlock() == nil {
+		return false, nil // networks field selects peers via CIDR blocks (IPs).
+		// nothing to do with Peer type which is not IPBlock
+	}
+	for _, cidr := range networks {
+		isIPv6, err := isIPv6Cidr(cidr)
+		if err != nil { // invalid cidr
+			return false, err
+		}
+		if isIPv6 { // not supported addresses
+			ruleWarnings = append(ruleWarnings, alerts.WarnUnsupportedIPv6Address)
+			continue // next cidr
+		}
+		ipb, err := netset.IPBlockFromCidr(string(cidr))
+		if err != nil {
+			return false, errors.New(netpolerrors.InvalidCIDRAddr)
+		}
+		if peer.GetPeerIPBlock().IsSubset(ipb) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
 // egressRuleSelectsPeer checks if the given []AdminNetworkPolicyEgressPeer rule selects the given peer
-// currently supposing a single egressPeer rule may contain only Namespaces/ Pods Fields,
-// @todo support also egress rule peer with Networks field
-// @todo if egress rule peer contains Nodes field, raise a warning that we don't support it
+// AdminNetworkPolicyEgressPeer may define an:
+// 1. in-cluster peer by one of  Namespaces/ Pods/ Nodes field
+// 2. an external peer with Networks field;
+// However, exactly one of the selector pointers must be set for a given peer.
 func egressRuleSelectsPeer(rulePeers []apisv1a.AdminNetworkPolicyEgressPeer, dst Peer) (bool, error) {
 	for i := range rulePeers {
-		fieldMatch, err := ruleFieldsSelectsPeer(rulePeers[i].Namespaces, rulePeers[i].Pods, dst)
+		// Validate: Exactly one of the `AdminNetworkPolicyEgressPeer`'s selector pointers must be
+		// set for a given peer.
+		if !validateEgressRuleFields(rulePeers[i]) {
+			return false, errors.New(netpolerrors.OneFieldSetRulePeerErr)
+		}
+		if rulePeers[i].Nodes != nil { // not supported field
+			ruleWarnings = append(ruleWarnings, alerts.WarnUnsupportedNodesField)
+			continue // next peer
```

Comment:
is there a test for this warning about nodes field?

Reply:
yes, tests/anp_and_banp_using_networks_and_nodes_test